v2/bigtable-cdc-to-hbase/src/test/java/com/google/cloud/teleport/v2/templates/BigtableToHbasePipelineIT.java

```diff
@@ -0,0 +1,329 @@
+import org.apache.hadoop.hbase.HBaseTestingUtility;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.util.Time;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** End to end table that runs the pipeline from Bigtable to Hbase. */
+@Category(TemplateIntegrationTest.class)
+@TemplateIntegrationTest(BigtableToHbasePipeline.class)
+@RunWith(JUnit4.class)
+public class BigtableToHbasePipelineIT extends TemplateTestBase {
+  private static final Logger LOG = LoggerFactory.getLogger(BigtableToHbasePipelineIT.class);
+
+  // private static StaticBigtableResourceManager bigtableResourceManager;
+  private static BigtableToHbasePipelineOptions pipelineOptions;
+  private static HBaseTestingUtility hBaseTestingUtility;
+  private static Table hbaseTable;
+  private static StaticBigtableResourceManager bigtableResourceManager;
+  private static String cbtQualifier;
+  private static String hbaseQualifier;
+  // Timeout for pipeline and tests
+  private static int testTimeoutSeconds = 30;
+
+  @BeforeClass
+  public static void setUpCluster() throws Exception {
+
+    // TODO: remove these params after Cdc GA
+    // Parse input as though from a live run. This requires passing in params to test via
+    // -Dparameters="..."
+    String input = System.getProperty("parameters");
+    String[] keyValuePairs = input.split(",");
+    Map<String, String> args = new HashMap<>();
+    for (String pair : keyValuePairs) {
+      String[] entry = pair.split("=");
+      args.put(entry[0], entry[1]);
+    }
+    hbaseQualifier = args.get("hbaseQualifier");
+    cbtQualifier = args.get("cbtQualifier");
+
+    // Create some pipeline options from args
+    pipelineOptions = PipelineOptionsFactory.create().as(BigtableToHbasePipelineOptions.class);
+    // Set bigtable change stream options
+    pipelineOptions.setBigtableProjectId(args.get("bigtableProjectId"));
+    pipelineOptions.setInstanceId(args.get("instanceId"));
+    pipelineOptions.setTableId(args.get("tableId"));
+    pipelineOptions.setAppProfileId(args.get("appProfileId"));
+    // Set bidirectional replication options
+    pipelineOptions.setBidirectionalReplicationEnabled(
+        Boolean.parseBoolean(args.get("bidirectionalReplicationEnabled")));
+    pipelineOptions.setCbtQualifier(cbtQualifier);
+    pipelineOptions.setHbaseQualifier(hbaseQualifier);
+
+    // Create Hbase cluster
+    hBaseTestingUtility = new HBaseTestingUtility();
+    hBaseTestingUtility.startMiniCluster();
+    // Create HBase table that mirrors persistent Hbase table.
+    hbaseTable = HbaseUtils.createTable(hBaseTestingUtility, pipelineOptions.getTableId());
```

Comment:
We have been creating an integration testing framework over the past few months that involves creating resource managers to manage resources used by templates. Since this template introduces a new resource (Hbase), this logic should be included in an hbase resource manager.

This may be something that needs to be added after this PR is submitted, but having custom resource management inside an IT is not something we plan to support.